src/olympia/devhub/tests/test_views_subscription.py

```diff
@@ -0,0 +1,282 @@
+import datetime
+
+from unittest import mock
+from waffle.testutils import override_switch
+
+from olympia.amo.tests import TestCase, addon_factory, user_factory
+from olympia.amo.urlresolvers import reverse
+from olympia.constants.promoted import VERIFIED
+from olympia.promoted.models import PromotedAddon, PromotedSubscription
+
+
+@override_switch("enable-subscription", active=True)
+class OnboardingSubscriptionTestCase(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.user = user_factory()
+        self.addon = addon_factory(users=[self.user])
+        self.promoted_addon = PromotedAddon.objects.create(
+            addon=self.addon, group_id=VERIFIED.id
+        )
+        self.subscription = PromotedSubscription.objects.filter(
+            promoted_addon=self.promoted_addon
+        ).get()
+        self.url = reverse(self.url_name, args=[self.addon.slug])
+        self.client.login(email=self.user.email)
+
+
+class TestOnboardingSubscription(OnboardingSubscriptionTestCase):
+    url_name = "devhub.addons.onboarding_subscription"
+
+    @override_switch("enable-subscription", active=False)
+    def test_returns_404_when_switch_is_disabled(self):
+        assert self.client.get(self.url).status_code == 404
+
+    def test_returns_404_when_subscription_is_not_found(self):
```

Comment:
Should we have a test where the subscription has a `stripe_session_id`, but `stripe.checkout.Session.retrieve()` cannot find it? It seems like we are assuming that a `stripe_session_id` will always correspond to a valid session, but isn't it possible it might not?

Reply:
maybe yes, although the current behavior is to return a 500. Maybe we should create a new session indeed.

Reply:
I added a test for that.